src/wallet/utils.rs

```diff
@@ -28,18 +26,19 @@ pub(crate) const SEQUENCE_LOCKTIME_MASK: u32 = 0x0000FFFF;
 // Threshold for nLockTime to be considered a block-height-based timelock rather than time-based
 pub(crate) const BLOCKS_TIMELOCK_THRESHOLD: u32 = 500000000;
 
-/// Trait to check if a value is below the dust limit
+/// Trait to check if a value is below the dust limit.
+/// We are performing dust value calculation for a given script public key using rust-bitcoin to
+/// keep it compatible with network dust rate
 // we implement this trait to make sure we don't mess up the comparison with off-by-one like a <
-// instead of a <= etc. The constant value for the dust limit is not public on purpose, to
-// encourage the usage of this trait.
+// instead of a <= etc.
 pub trait IsDust {
     /// Check whether or not a value is below dust limit
-    fn is_dust(&self) -> bool;
+    fn is_dust(&self, script: &Script) -> bool;
 }
 
 impl IsDust for u64 {
-    fn is_dust(&self) -> bool {
-        *self <= DUST_LIMIT_SATOSHI
+    fn is_dust(&self, script: &Script) -> bool {
+        *self <= script.dust_value().as_sat()
```

Comment:
I think this should be a lt, not lte: from the [docs in rust-bitcoin](https://docs.rs/bitcoin/latest/bitcoin/blockdata/script/struct.Script.html#method.dust_value):

> Gets the minimum value an output with this script should have in order to be broadcastable on today’s bitcoin network

I interpret this as "if the script has this value, then it is broadcastable", so it should be considered dust if it *does not* have this value, hence only if it's lower than it.

Ideally we should also have a test for this somewhere with known good/bad values

Reply:
Sorry for the late response. Just now seeing this! Good find--I think your interpretation is correct, and I will correspondingly fix it. I will also try to find some good/bad values to construct a test.